File: bot/src/main/java/com/yauhenl/pibot/service/BotControl.java
package com.yauhenl.pibot.service;

import com.pi4j.io.gpio.GpioFactory;
import com.yauhenl.pibot.hardware.cascade.BotCascade;
import com.yauhenl.pibot.hardware.cascade.Motor;
import com.yauhenl.pibot.hardware.cascade.MotorKey;
import com.yauhenl.pibot.hardware.cascade.Shift;

import java.util.Map;

/**
 * Created by Stsiapan_Shablinski on 7/29/2016.
 */
public class BotControl {

    private BotCascade cascade;
    private static BotControl botControl = new BotControl();

    private BotControl() {
        cascade = new BotCascade();
        cascade.initCascade(GpioFactory.getInstance());
    }

    public static BotControl getInstance() {
        return botControl;
    }

    public Shift getShift() {
        return cascade.getShift();
    }

    public Map<MotorKey, Motor> getMotors() {
        return cascade.getMotors();
    }
}

File: bot/src/main/kotlin/com/yauhenl/pibot/service/BotControl.kt
package com.yauhenl.pibot.service

import com.pi4j.io.gpio.GpioFactory
import com.yauhenl.pibot.hardware.BotCascade
import com.yauhenl.pibot.hardware.Motor
import com.yauhenl.pibot.hardware.MotorKey
import com.yauhenl.pibot.hardware.ShiftRegister
import org.springframework.stereotype.Service

@Service
class BotControl {

    private val cascade: BotCascade = BotCascade(GpioFactory.getInstance())

    val shiftRegister: ShiftRegister = cascade.shiftRegister

    val motors: Map<MotorKey, Motor> = cascade.motors
}
